modules/fedimint-ln/src/db.rs

```diff
@@ -118,3 +118,143 @@ impl_db_lookup!(
     key = LightningGatewayKey,
     query_prefix = LightningGatewayKeyPrefix
 );
+
+#[cfg(test)]
+mod fedimint_migration_tests {
+    use std::collections::BTreeMap;
+    use std::env;
+    use std::path::Path;
+
+    use fedimint_core::core::LEGACY_HARDCODED_INSTANCE_ID_LN;
+    use fedimint_core::db::apply_migrations;
+    use fedimint_core::module::DynModuleGen;
+    use fedimint_testing::apply_to_databases;
+    use futures::StreamExt;
+    use strum::IntoEnumIterator;
+
+    use crate::db::{
+        AgreedDecryptionShareKeyPrefix, ContractKeyPrefix, ContractUpdateKeyPrefix, DbKeyPrefix,
+        LightningGatewayKeyPrefix, ProposeDecryptionShareKeyPrefix,
+    };
+    use crate::LightningGen;
+
+    #[tokio::test(flavor = "multi_thread")]
+    async fn verify_lightning_database_migration() {
+        // Only run the database migration test if the database backup directory
+        // environment variable has been defined
+        let backup_dir = env::var("FM_TEST_DB_BACKUP_DIR");
+        if let Ok(backup_dir) = backup_dir {
+            let mut migrated_values = BTreeMap::new();
+            apply_to_databases(
+                Path::new(&backup_dir),
+                |db| async move {
+                    // First apply all of the database migrations so that the data can be properly
+                    // read.
+                    let module = DynModuleGen::from(LightningGen);
+                    let isolated_db = db.new_isolated(LEGACY_HARDCODED_INSTANCE_ID_LN);
+                    apply_migrations(
+                        &isolated_db,
+                        module.module_kind().to_string(),
+                        module.database_version(),
+                        module.get_database_migrations(),
+                    )
+                    .await
+                    .expect("Error applying migrations to temp database");
+
+                    // Verify that all of the data from the lightning namespace can be read. If a
+                    // database migration failed or was not properly supplied,
+                    // this will fail.
+                    let mut migrated_pairs: BTreeMap<u8, usize> = BTreeMap::new();
+                    let mut dbtx = isolated_db.begin_transaction().await;
+
+                    for prefix in DbKeyPrefix::iter() {
```

Comment:
Move to another function maybe to make more readable?

Reply:
Do you think this is still needed? A macro might help with readability here, similar to what we did in `dbtool`